modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,374 @@
+                break;
+
+            case PacketType::Data:
+                SAMPLE_TYPE_DISPATCH(inputSampleType, processDataPacket, packet);
+                break;
+
+            default:
+                break;
+        }
+
+        packet = connection.dequeue();
+    };
+}
+
+void ClassifierFbImpl::processEventPacket(const EventPacketPtr& packet)
+{
+    if (packet.getEventId() == event_packet_id::DATA_DESCRIPTOR_CHANGED)
+    {
+        DataDescriptorPtr inputDataDescriptor = packet.getParameters().get(event_packet_param::DATA_DESCRIPTOR);
+        DataDescriptorPtr inputDomainDataDescriptor = packet.getParameters().get(event_packet_param::DOMAIN_DATA_DESCRIPTOR);
+        processSignalDescriptorChanged(inputDataDescriptor, inputDomainDataDescriptor);
+    }
+}
+
+inline UInt ClassifierFbImpl::timeMs(UInt time) 
+{
+    return time * inputDeltaTicks;
+}
+
+inline bool ClassifierFbImpl::timeInInterval(UInt startTime, UInt endTime) 
+{
+    return (endTime - startTime) < timeMs(blockSizeMs);
+}
+
+template <SampleType InputSampleType>
+void ClassifierFbImpl::processDataPacket(const DataPacketPtr& packet)
+{
+    using InputType = typename SampleTypeToType<InputSampleType>::Type;
+    using OutputType = Float;
+
+    // if packet does not have samples - ignore
+    if (packet.getSampleCount() == 0)
+        return;
+
+    packets.push_back(packet);
+
+    size_t outputPackages = 0;
+    {
+        auto inputDomainData = static_cast<UInt*>(packet.getDomainPacket().getData());
+        UInt lastTime = inputDomainData[packet.getSampleCount() - 1];
+
+        // initialize members
+        if (packetStarted == UInt{})
+        {
+            packetStarted = inputDomainData[0];
+            sampleStarted = 0; 
+        }
+
+        outputPackages = (lastTime - packetStarted) / timeMs(blockSizeMs);
+    }
+
+    if (!outputPackages)
+        return;
+
+    auto outputDomainPacket = DataPacket(outputDomainDataDescriptor, 1);
+    auto outputDomainData = static_cast<UInt*>(outputDomainPacket.getData());
+
+    DataPacketPtr outputPacket {};
+    OutputType* outputData {};
+
+    auto rangeSize = outputDataDescriptor.getDimensions()[0].getSize();
+    auto labels = outputDataDescriptor.getDimensions()[0].getLabels();
+    if (labels.getCount() == 0) 
+        return;
+
+    size_t packageVals = 0;
+    while (outputPackages && packets.size()) 
+    {
+        auto listPacket = packets.front();
+
+        auto sampleCnt = listPacket.getSampleCount();
+        auto inputData = static_cast<InputType*>(listPacket.getData());
+        auto inputDomainData = static_cast<UInt*>(listPacket.getDomainPacket().getData());
+       
+        // reset array for new package
+        if (packageVals == 0)
+        {
+            outputPacket = DataPacketWithDomain(outputDomainPacket, outputDataDescriptor, 1);
+            outputData = static_cast<OutputType*>(outputPacket.getData());
+            memset(outputData, 0, rangeSize * sizeof(OutputType));
```

Comment:
This is invalid if we are using explicit dimensions, as rangeSize should be equal to the dimension size.